### DISCUSSION TITLE
When the result of LogQL is "no data", even though the alert state is "Normal", Grafana sends an alert.

### DISCUSSION THREAD

**Discussion by nutslove** [grafana/grafana#47271] [Alerting]
I have an issue with alerts from grafana based on loki.
When the result of LogQL is "no data", even though the alert state is "Normal", Grafana sends an alert.
Is this the way it is?
I think It won't happen if I can turn no data to zero in LogQL, is there a way to turn no data to zero?

![image (6)](https://user-images.githubusercontent.com/43600385/161561776-fc54608c-7023-49ee-9246-abf594152813.png)
![image (5)](https://user-images.githubusercontent.com/43600385/161561780-3db01118-bc77-42ff-a527-248106f26eb8.png)
![image (7)](https://user-images.githubusercontent.com/43600385/161561785-3eaa1e1c-a594-415f-8a3f-6b38276e534a.png)

**Comment by AhHa45** [comment]
bump

**Reply by timam** [reply]
Is there any way around?

**Comment by jimhol** [comment]
Bump. I seem to be getting alerts telling me there have been no errors. This is not a useful alert.

I have "Alert state if no data or all values are null" as "OK". Still getting these.

**Comment by michaelr524** [comment]
Any clues?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
